server/src/test/java/org/elasticsearch/index/mapper/GeoShapeFieldMapperTests.java

```diff
@@ -219,7 +229,138 @@ protected Object generateRandomInputValue(MappedFieldType ft) {
 
     @Override
     protected SyntheticSourceSupport syntheticSourceSupport(boolean ignoreMalformed) {
```

Comment:
I started adding tests here because it looks like the most complex type. I am not sure how should we proceed - if this PR contains tests for all fields that now have support, it will be quite big.